app/routes/enterprise.py

```diff
@@ -89,4 +99,65 @@ async def create_enterprise(
                 status_code=HTTP_409_CONFLICT,
             )
 
+    @enterprise_router.patch(
+        "/enterprise/{enterprise_id}",
+        status_code=200,
+        response_model=EnterpriseUpdateResponse,
+        responses={**get_verify_enterprise_permissions_responses()}
+    )
+    async def update_enterprise(
+            enterprise_id: int,
+            item: EnterpriseUpdateResponse,
+            user: models.User = Depends(CurrentUser())
+    ):
+        enterprise = await models.Enterprise.objects.get_or_none(id=enterprise_id)
+        if not enterprise:
+            raise HTTPException(status_code=404, detail=f"Invoice {enterprise_id} not found")
+
+        permissions = await verify_enterprise_permissions(
+            user,
+            enterprise_id,
+            required_permissions=[
+                models.UserEnterpriseRoles.admin,
+            ]
+        )
+        if permissions is True:
+            update_data = item.dict(exclude_unset=True)
+            enterprise.update(**update_data)
+            enterprise_output = EnterpriseResponse(
+                enterprise_id=enterprise.id,
+                name=enterprise.name,
+                role="ADMIN",
+                nip_number=enterprise.nip_number,
+                address=enterprise.address,
+            )
+            return enterprise_output
+        return permissions
+
+    @enterprise_router.delete(
+        "/enterprise/{enterprise_id}",
+        status_code=200,
```

Comment:
It needs response model - frontend build fails if there is no response model

Reply:
Fair enough, I'll apply the same to invoice